LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2DownloadableContentInfo_LW_FactionBalance.uc

```diff
@@ -346,6 +348,52 @@ static function EditParry(X2AbilityTemplate Template)
 	PersistentEffect.SetDisplayInfo(ePerkBuff_Passive, ParryTemplate.LocFriendlyName, ParryTemplate.GetMyHelpText(), ParryTemplate.IconImage, true, , ParryTemplate.AbilitySourceName);
 	ParryTemplate.AddTargetEffect(PersistentEffect);
 }
+//Copy pasted Realitymachina's code
+static function EditReward()
+{
+	local X2RewardTemplate RewardTemplate;
+	local X2StrategyElementTemplateManager StratMgr;
+	StratMgr = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+
+
+	RewardTemplate = X2RewardTemplate(StratMgr.FindStrategyElementTemplate('Reward_ExtraFactionSoldier'));
+
+	if(RewardTemplate != none)
+		RewardTemplate.IsRewardAvailableFn = IsExtraSoldierAvailable;
+}
+
+static function bool IsExtraSoldierAvailable(optional XComGameState NewGameState, optional StateObjectReference AuxRef)
+{
+	local XComGameState_ResistanceFaction FactionState;
+	local int NumFactionSoldiers;
+	local XComGameState_CovertAction ActionState;
+	local XComGameStateHistory History;
+
+	History = `XCOMHISTORY;
+
+
+
+	FactionState = class'X2StrategyElement_DefaultRewards'.static.GetFactionState(NewGameState, AuxRef);
+	if (FactionState != none)
+	{
+
+
+		NumFactionSoldiers = FactionState.GetNumFactionSoldiers(NewGameState);
+		foreach History.IterateByClassType(class'XComGameState_CovertAction', ActionState)
+		{
+			if(ActionState.GetMyTemplateName() == 'CovertAction_RecruitExtraFactionSoldier' && (ActionState.GetFaction().GetReference().ObjectID == FactionState.GetReference().ObjectID) && ActionState.bStarted && !ActionState.bCompleted) //this is dumb but we have to account for this
+				NumFactionSoldiers += 1;
+		}

```

Comment:
I don't understand why this code is necessary. It seems like a sledgehammer solution. Is there a bug in the base game it's trying to get round?

Reply:
This shouldn't be needed. The community highlander has a fix for duplicate covert actions appearing while one is already running. So I'd prefer to remove this hack on the assumption it's not needed. If we need to fix it later on, so be it.